### PR TITLE
Entity added, properties and first methods added

### PR DESCRIPTION
**Pull request by azzilian**
Pom - added dependencies for springboot,mysql,lombok
application.properties - added properties to connect to DB OnlineBookStoreApplication - added simple test
Book - entity added
BookService + BookRepository - methods save + findAll ad

### CODE REVIEW

File: src/main/java/com/onlinebookstore/onlinebookstore/OnlineBookStoreApplication.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class OnlineBookStoreApplication {
+    @Autowired
+    private BookService bookService;
+
+    public static void main(String[] args) {
+        SpringApplication.run(OnlineBookStoreApplication.class, args);
+    }
```
Comment (andrii-hoienko) [inline src/main/java/com/onlinebookstore/onlinebookstore/OnlineBookStoreApplication.java:19]: Is it necessary?

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.onlinebookstore.onlinebookstore.model.Book;
+import java.util.List;
+
+public interface BookRepository {
+
+    Book save(Book book);
+
+    List<Book> findAll();
+
```
Comment (andrii-hoienko) [inline src/main/java/com/onlinebookstore/onlinebookstore/repository/BookRepository.java]:
```suggestion

```

File: src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book implements Cloneable {
```
Comment (andrii-hoienko) [inline src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java]: What is the reason to make of clonable?

File: src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Id
+    private Long id;
+
+    @NotBlank
+    @Size(max = 255)
```
Comment (andrii-hoienko) [inline src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java]: I believe validations should be on dto

File: src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> query = session.createQuery("FROM Book", Book.class);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can not find any books ", e);
```
Comment (andrii-hoienko) [inline src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can not find any books ", e);

```

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1 +1,7 @@
-spring.application.name=onlinebookstore
+spring.datasource.url=jdbc:mysql://localhost/bookstoreDB?serverTimezone=UTC
+spring.datasource.username=root
+spring.datasource.password=12345678
```
Comment (andrii-hoienko) [inline src/main/resources/application.properties]: Better not to push credentials on  remote, left these fields blank

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job, but some improvements required

File: src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java (hunk 3 above)
Comment (azzilian) [reply src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java]: It a leftovers from first raw version, I will delete it

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**